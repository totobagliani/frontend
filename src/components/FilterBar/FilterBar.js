import React from 'react';

export default function FilterBar() {
  return (
    <div>
      <h1>FilterBar</h1>
    </div>
  );
}
