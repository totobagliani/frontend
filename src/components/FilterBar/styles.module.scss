@import '../../styles/settings.scss';

.filterbar {
  background-color: antiquewhite;
}
.filterbar__form {
  display: flex;
  justify-content: center;

  border-bottom: 1px solid $gray-normal;
}

.filterbar__select:first-child {
  border-right: 1px solid $gray-normal;
}

.filterbar__select {
  outline-style: none;
  border: none;
  width: 50%;
  padding: 1em;
  font-family: $orderPaginationFont;
  font-size: 1.3em;
  font-weight: bold;
  line-height: 32px;
}
