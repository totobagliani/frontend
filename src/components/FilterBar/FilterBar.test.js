/* eslint-disable comma-dangle */
import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import userEvent from '@testing-library/user-event';
import configureStore from '../../redux/store';
import { setFilter, setOrder } from '../../redux/actions/ui.action';
import FilterBar from './FilterBar';

const initialState = {
  products: {
    products: [],
    productAdded: {
      productName: '',
      description: '',
      imageURL: '',
      isFavourite: false,
      price: 0,
      section: ''
    },
    searchResults: {
      itemsCount: 0,
      searchTerm: ''
    }
  },
  ui: {
    currentPage: 1,
    currentFilter: '',
    currentOrder: '',
    resultsVisible: false
  }
};
const store = configureStore(initialState);

store.dispatch = jest.fn();

jest.mock('../../redux/actions/ui.action', () => ({
  setCurrentPage: jest.fn(),
  setFilter: jest.fn(),
  setOrder: jest.fn()
}));

describe('Given the FilterBar Component', () => {
  beforeEach(() => {
    render(
      <Provider store={store}>
        <BrowserRouter>
          <FilterBar />
        </BrowserRouter>
      </Provider>
    );
  });
  afterEach(() => {
    jest.clearAllMocks();
  });
  test('then must be rendered (take a snapshot)', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <BrowserRouter>
          <FilterBar />
        </BrowserRouter>
      </Provider>
    );
    expect(asFragment(<FilterBar />)).toMatchSnapshot();
  });
  test('when select a section in Select then call to handleSetFilter', () => {
    /*  const selectItems = screen.getByRole('combobox', {
      name: 'product-section'
    }); */
    const selectProduct = screen.getByDisplayValue(/productos/i);
    const optionCocina = screen.getByText(/Cocina - 0/i);

    userEvent.selectOptions(selectProduct, optionCocina);
    expect(optionCocina.value).toBe('Cocina');
    expect(setFilter).toHaveBeenCalled();
  });
  test('when select a order in Select then call to handleSetOrder', () => {
    const selectOrder = screen.getByDisplayValue(/Ordenar Por/i);
    const optionPrecio = screen.getByText(/precio - 0/i);
    userEvent.selectOptions(selectOrder, optionPrecio);
    expect(optionPrecio.value).toBe('precio');
    expect(setOrder).toHaveBeenCalled();
  });
});
