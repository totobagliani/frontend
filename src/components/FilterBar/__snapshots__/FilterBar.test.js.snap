// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given the FilterBar Component then must be rendered (take a snapshot) 1`] = `
<DocumentFragment>
  <div
    class="filterbar"
  >
    <form
      class="filterbar__form"
    >
      <select
        class="filterbar__select"
        name="product-section"
      >
        <option
          class="filterbar__optiontitle"
          selected=""
          value=""
        >
           Productos (2)
        </option>
        <option
          class="filterbar__option"
          value="Ba침o"
        >
          Ba침o - 0
        </option>
        <option
          class="filterbar__option"
          value="Dormitorio"
        >
          Dormitorio - 0
        </option>
        <option
          class="filterbar__option"
          value="Cocina"
        >
          Cocina - 0
        </option>
        <option
          class="filterbar__option"
          value="Salon"
        >
          Salon - 0
        </option>
        <option
          class="filterbar__option"
          value="Terraza"
        >
          Terraza - 0
        </option>
        <option
          class="filterbar__option"
          value="Comedor"
        >
          Comedor - 0
        </option>
      </select>
      <select
        class="filterbar__select"
        name="order"
      >
        <option
          class="filterbar__optiontitle"
          selected=""
          value=""
        >
          Ordenar Por:
        </option>
        <option
          class="filterbar__option"
          value="Precio"
        >
          Precio
        </option>
        <option
          class="filterbar__option"
          value="Nombre"
        >
          Nombre
        </option>
        <option
          class="filterbar__option"
          value="Secci칩n"
        >
          Secci칩n
        </option>
      </select>
    </form>
  </div>
</DocumentFragment>
`;
